wingtips-zipkin/src/main/java/com/nike/wingtips/zipkin/util/ZipkinSpanSenderDefaultHttpImpl.java

```diff
@@ -0,0 +1,203 @@
+     *                               e.g. {@code http://localhost:9411}, or {@code https://zipkinserver.doesnotexist.com/}
+     * @param compressZipkinSpanPayload Pass in true if the payload sent to the Zipkin server should be gzipped, false to pass the payload uncompressed.
+     */
+    public ZipkinSpanSenderDefaultHttpImpl(String postZipkinSpansBaseUrl, boolean compressZipkinSpanPayload) {
+        this(postZipkinSpansBaseUrl, compressZipkinSpanPayload, DEFAULT_CONNECT_TIMEOUT_MILLIS, DEFAULT_READ_TIMEOUT_MILLIS,
+             DEFAULT_SPAN_BATCH_SENDING_PERIOD_MILLIS);
+    }
+
+    @Override
+    public void handleSpan(zipkin.Span span) {
+        zipkinSpanSendingQueue.offer(span);
+    }
+
+    @Override
+    public void forcePurgeBatchedSpans() {
+        zipkinSpanSendingScheduler.execute(senderJob);
+    }
+
+    protected ScheduledExecutorService configureScheduledExecutorServiceForBatching() {
+        return Executors.newSingleThreadScheduledExecutor(new ThreadFactory() {
+            @Override
+            public Thread newThread(final Runnable r) {
+                return new Thread(r, "zipkin-span-sender");
+            }
+        });
+    }
+
+    protected void sendSpans(List<zipkin.Span> spanList) {
+        try {
+            sendSpans(Codec.JSON.writeSpans(spanList));
+        } catch (IOException e) {
+            Set<String> affectedTraceIds = new HashSet<>(spanList.size());
+            for (zipkin.Span span : spanList) {
+                affectedTraceIds.add(String.valueOf(span.traceId));
+            }
+            logger.error("An error occurred attempting to post Zipkin spans to the Zipkin server. affected_trace_ids={}, exception_cause=\"{}\"",
+                         affectedTraceIds.toString(), e.toString());
+        }
+    }
+
+    /**
+     * <p>
+     *     This method uses basic JDK classes to POST the given payload bytes (representing a list of Zipkin Spans that have been serialized to JSON)
+     *     to the Zipkin server endpoint at {@link #postZipkinSpansUrl}. This gives a simple workable implementation that requires no outside dependencies,
+     *     however it may lack some of the flexibility your project requires (e.g. around connection pooling). You can extend this class and override
+     *     this method to use a different HTTP client for sending spans to Zipkin if desired.
+     * </p>
+     * <p>
+     *     This code was derived from the Zipkin/Brave repository's brave-spancollector-http module v3.9.1
+     *     (https://github.com/openzipkin/brave/blob/master/brave-spancollector-http/src/main/java/com/github/kristofa/brave/http/HttpSpanCollector.java)
+     *     and licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0).
+     * </p>
+     */
+    protected void sendSpans(byte[] spanListJsonPayloadBytes) throws IOException {
```

Comment:
I wanted to avoid pulling in brave. Maybe there's no good reason for that, I'm not sure. But I *would* use the zipkin-reporter-java stuff you're looking at creating since it's focused specifically on the send-spans-to-zipkin use case.